geopandas/tests/test_pandas_methods.py

```diff
@@ -124,13 +124,34 @@ def test_assign(df):
     assert_frame_equal(res, exp, )
 
 
-def test_astype(s):
+def test_astype(s, df):
+
+    # check geoseries functionality
 
     with pytest.raises(TypeError):
         s.astype(int)
 
     assert s.astype(str)[0] == 'POINT (0 0)'
 
+    # check whether returned object is a geodataframe
+
+    df.rename_geometry('geom_list', True)
+    df = df.astype({'value1':float})
+
+    assert isinstance(df, GeoDataFrame)
+
+    # check that a different geometry column name is OK
+
+    try:
+        df.geometry
+    except AttributeError:
+        pytest.fail('No geometry column found')
+
+    assert df.geometry.astype(str)[0] == 'POINT (0 0)'
```

Comment:
This is very similar to the test above for GeoSeries, so I would rather test something like:

```
df.astype(str)
df.astype({'geom_list': str'})
```

and check for both cases that is is not a GeoDataFrame.

Reply:
For some reason the first statement `df.astype(str)` still returns a GeoDataFrame. This occurs within the newly implemented `df = super(DataFrame, self).astype(dtype, copy=copy, errors=errors, **kwargs)`. The second statement returns a `DataFrame` as expected.

Reply:
I commented above for a fix for this problem